### PR TITLE
feat: implement public registry alias

### PR DESCRIPTION
**Pull request by brivu**
This `PR` implements the `public-registry-alias` parameter, enabling customers to push an image to a public `ecr` registry.

### CODE REVIEW

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:f2317dbc9f4f60f14d71ac9115cf7041fff0780b

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:04cc772cb9c80fb65dd6a2f0d47b026d621286e5

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:8b2ea53b991039cca671f8585fc5225e4588755f

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:510cbdd902833f3894864bbdc6f8beecc0dff587

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:164967ac035a63bdf3fec83d58f275a4fbfc105a

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:0a7c01f8e5f48ef693aab9180d88a95f76b14e31

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:22368edca0cd39336927979b9d43d05b5844d693

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:22368edca0cd39336927979b9d43d05b5844d693

File: .circleci/test-deploy.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 version: 2.1
 orbs:
   aws-ecr: circleci/aws-ecr@dev:<<pipeline.git.revision>>
-  orb-tools: circleci/orb-tools@11.1
+  orb-tools: circleci/orb-tools@11.3
```
Comment (Jaryt) [inline .circleci/test-deploy.yml:4]: Orb tools is on .4 now :)

File: src/commands/build-image.yml [hunk 2]
```diff
@@ -95,5 +102,6 @@ steps:
         PARAM_PUBLIC_REGISTRY: <<parameters.public-registry>>
         PARAM_PUSH_IMAGE: <<parameters.push-image>>
         PARAM_LIFECYCLE_POLICY_PATH: <<parameters.lifecycle-policy-path>>
+        PARAM_PUBLIC_REGISTRY_ALIAS: <<parameters.public-registry-alias>>
```
Comment (Jaryt) [inline src/commands/build-image.yml:105]: Check out the Orb standards, we've defined them for parameter naming.

**Review by Jaryt — CHANGES_REQUESTED** [review]
Looking good. Some small changes needed!

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:a83d5142574ab57a7e38eaacf5552b0e7bee3334

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:80083fb4a90e4659daf2c1f4bced95612b502040

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:54b5114701694111732a9dffb69e2afafc37b54e

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:fd67de25d6fb850799cd196a5fe88dc56f00fb6f

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:589b226f95d4e3c03c25ad079d35e25c7946ff02

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:025236066b39197eb8031ca1b2e2a92381458fdf

File: src/scripts/ecr-login.sh [hunk 3]
```diff
@@ -1,21 +1,21 @@
 #!/bin/bash
-PARAM_REGION=$(eval echo "${PARAM_REGION}")
-PARAM_ACCOUNT_URL="${!PARAM_REGISTRY_ID}.dkr.ecr.${PARAM_REGION}.amazonaws.com"
+ORB_EVAL_REGION=$(eval echo "${ORB_EVAL_REGION}")
+ORB_VAL_ACCOUNT_URL="${!ORB_EVAL_REGISTRY_ID}.dkr.ecr.${ORB_EVAL_REGION}.amazonaws.com"
```
Comment (Jaryt) [inline src/scripts/ecr-login.sh:3]:
```suggestion
ORB_VAL_ACCOUNT_URL="${!ORB_ENV_REGISTRY_ID}.dkr.ecr.${ORB_EVAL_REGION}.amazonaws.com"
```

File: src/commands/ecr-login.yml [hunk 4]
```diff
@@ -116,10 +116,10 @@ steps:
         - run:
             name: Log into Amazon ECR with profile <<parameters.new-profile-name>>
             environment:
-              PARAM_PROFILE_NAME: <<parameters.new-profile-name>>
-              PARAM_REGISTRY_ID: <<parameters.registry-id>>
-              PARAM_REGION: <<parameters.region>>
-              PARAM_PUBLIC_REGISTRY: <<parameters.public-registry>>
+              ORB_VAL_PROFILE_NAME: <<parameters.new-profile-name>>
+              ORB_VAL_REGISTRY_ID: <<parameters.registry-id>>
```
Comment (Jaryt) [inline src/commands/ecr-login.yml:120]:
```suggestion
              ORB_ENV_REGISTRY_ID: <<parameters.registry-id>>
```

For parameters type of `env_var_name` the prefix should be `ORB_ENV_`. This indicates later on that you should indirectly reference the environment variable.

**Review by Jaryt — CHANGES_REQUESTED** [review]
Getting there! Please do a pass and check on `env_var_name` types

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:fd37e3ee01ce0fe84e76f0872ad0356105a71f40

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:041f955fe36eda96f9e9079e2392ac073fe6e944

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=dev:5422f5a0077dd6e449b5a55f0cd15cbfcf78ebba

**Review by Jaryt — APPROVED** [review]
Good work!

**Comment by JeroenRijks** [comment]
Hi, when will this PR make it into a published version of the orb? My org is currently having issues caused by the https://github.com/CircleCI-Public/aws-ecr-orb/issues/227 bug - if this could get released as 8.1.4, that would be really useful!

**Comment by Jedv-TransReport** [comment]
> Hi, when will this PR make it into a published version of the orb? My org is currently having issues caused by the #227 bug - if this could get released as 8.1.4, that would be really useful!

I second this! Really needed.

**Pull request merged**